### ISSUE TITLE
create custom `Config` objects easier

### ISSUE THREAD

**Issue by tannernelson** [vapor/vapor#532]
Currently, you can only seed `Config` objects with an `app` file.

It would be really nice for parity and testing if we could create `Config` objects from a full tree of JSON. This would allow, for example, to plug in test `mysql.json` values in the `mysql-provider` tests.

- [ ] Allow custom Config object that can encapsulate all files (totally displace Config/ directory)
- [ ] Use file functions from Core and move any left over `FileManager` stuff to Core
- [ ] Load environment variables at the time the Config is initialized (will probably require recursion through the Node)

**Comment by tannernelson** [comment]
For instance:

```swift
let config = Config([
	"servers": [
		"default": [
			"port": "$PORT"
		]
	]
])

let drop = Droplet(config: config)
```

@IanKeen

**Comment by LoganWright** [comment]
closed w/ #552 

Config is now considerably more flexible, and completely definable in memory in customizable ways.
